protect_archiver/cli/download.py

```diff
@@ -0,0 +1,184 @@
+import click
+
+from datetime import datetime
+from os import path
+
+from .base import cli
+from ..client import ProtectClient, ProtectError
+
+
+@cli.command("download", help="Download footage from a local UniFi Protect system")
+@click.argument("dest")
+@click.option("--address", default="unifi", help="CloudKey IP address or hostname")
+@click.option("--port", default=7443, help="UniFi Protect service port")
+@click.option(
+    "--username", default="ubnt", help="Username of user with local access",
+)
+@click.option(
+    "--password", required=True, help="Password of user with local access",
+)
+@click.option(
+    "--verify-ssl", default=False, help="Verify CloudKey SSL certificate",
+)
+@click.option(
+    "--cameras",
+    default="all",
+    help=(
+        "Comma-separated list of one or more camera IDs ('--cameras=\"id_1,id_2,id_3,...\"'). "
+        "Use '--cameras=all' to download footage of all available cameras."
+    ),
+)
+@click.option(
+    "--wait-between-downloads",
+    "download_wait",
+    default=0,
+    help="Time to wait between file downloads, in seconds (Default: 0)",
+)
+@click.option(
+    "--downloads-before-key-refresh",
+    "max_downloads_with_key",
+    default=3,
+    help="Maximum number of downloads with the same API Access Key (Default: 3)",
+)
+@click.option(
+    "--downloads-before-auth-refresh",
+    "max_downloads_with_auth",
+    default=10,
+    help="Maximum number of downloads with the same API Authentication Token (Default: 10)",
+)
+@click.option(
+    "--ignore-failed-downloads",
+    is_flag=True,
+    default=False,
+    help="Ignore failed downloads and continue with next download (Default: False)",
+)
+@click.option(
+    "--skip-existing-files",
+    is_flag=True,
+    default=False,
+    help="Skip downloading files which already exist on disk (Default: False)",
+)
+@click.option(
+    "--touch-files",
+    default=False,
+    is_flag=True,
+    help=(
+        "Create local file without content for current download (Default: False) - "
+        "useful in combination with '--skip-existing-files' to skip problematic segments"
+    ),
+)
+@click.option(
+    "--use-subfolders/--no-use-subfolders",
+    default=True,
+    help="Save footage to folder structure with format 'YYYY/MM/DD/camera_name/' (Default: False)",
+)
+@click.option(
+    "--download-request-timeout",
+    "download_timeout",
+    default=60,
+    help="Time to wait before aborting download request, in seconds (Default: 60)",
+)
+@click.option(
+    "--start",
+    required=False,
+    help='Start time in dateutil.parser compatible format, for example "YYYY-MM-DD HH:MM:SS+0000"',
+)
+@click.option(
+    "--end",
+    type=str,
+    required=False,
+    help='End time in dateutil.parser compatible format, for example "YYYY-MM-DD HH:MM:SS+0000"',
+)
+@click.option(
```

Comment:
we could move snapshot into its own command to separate logic even more at this point

Reply:
agreed since it's not directly part of the download/archive process